### DISCUSSION TITLE
Explanation Quote

### DISCUSSION THREAD

**Discussion by fxmb** [medusajs/b2b-starter-medusa#8] [Q&A]
Can you guys explain the quoting feature a bit? We built this internally and maybe we can contribute some thoughts and gotchas beforehand from a business logic. 

Would like to better understand how exactly you define "quotes". It's a huge topic for us. 

Thanks for your time guys.

**Comment by riqwan** [comment]
Hey @fxmb,

The goal of the quotes in the starter is to have a common base workflow running:

1. customer can request quote
2. merchant can provide quote
  a. add items
  b. update price
  c. remove items 
4. customer can reject or accept quote
5. merchant can reject quote
6. convert quote to order

People using the starter can then easily customize it to their needs as I can tell that there are multiple ways to go about quotes. Can you share how you went about building it and what gotchas you came across?

**Reply by fxmb** [reply]
Wow this is insane... We need just that

**Reply by fxmb** [reply]
Here are some thoughts around this:

1. Quotes usually involve articles that are not instantly deliverable (at least in our experience and others). There should be a delivery date involved. That's why customers usually request a quote since they cannot buy it immediately. 
2. There should be a validation date with a default. Eg quote is valid for 30 days. 
3. There should be a team member of the shop as the responsible person. Basically an admin user that is assigned to a quote. 
4. There should be an option to send the quote quickly via email as pdf (probably should be done via custom integration)
5. Send follow up for quote via button => "hey are you still interested in this?"
6. We are thinking of introducing automatic quote ratios. Eg: "hey, you have asked for 20 quotes in the past 14 days but never bought. What is wrong? Price too high etc?"
7. Overall internal dashboard for quotes to track the status in admin. Ideally a ticket system using some linear style canban? States could be draft, pending, won, lost, cancelled etc. 

I know this is a lot and some of it is up to the user to develop I believe but maybe some inspiration

**Reply by fxmb** [reply]
Also:
We run into the problem of offering articles we do not have in the shop. Think of a unique adidas shoe that we could theoretically get but we do not have it in the store listed. Happens daily.

**Reply by fxmb** [reply]
Plus we introduced the feature that customers can download quotes for themselves and it's extremely popular. Basically in the cart you can press "download cart as quote" and you get a pdf. This is used by many customers of big companies who need approval from other teams before making a purchase (finance team needs to approve the expense eg). 
Would be awesome to have that by default. Also: "send quote via email" which is something we are implementing in the future potentially

**Reply by OD91** [reply]
Hey everyone! Just wanted to add a few more things to consider for the quote process:

1. **Company Info on Both Sides**: It’s pretty standard for customers to need both the issuer’s company address and their own (including a - optional - contact person) on the quote. So, no blank headers or just the issuer info—make sure the recipient is there too.
2. **Offer ID:** Super important to have an offer ID on there as a reference. Makes it easier for everyone to track things.
3. **Shipping Costs**: Shipping costs should be listed as a separate line item. Also, somewhere we should mention the **delivery address**—sometimes the quote comes from a company in one country, but the delivery is to another.
4. **Delivery Options**: We should also add a field for delivery options like Express or Standard.
5. **Line Item Comments**: Adding a comment field for each line item would be handy. For example, “This is the successor of article X.”
6. **Payment Terms**: And of course, payment terms! Could be something like "upfront via proforma invoice," or "30 days net / 10 days 2%", etc.

**Reply by riqwan** [reply]
Thanks guys, these are great suggestions! We can potentially pick up some of these if we have some time to build them after we have achieved our goals internally. 

Looking forward to this. 🔥
